#1 : 프로젝트 초기화 및 테스트 컨트롤러 생성

tlqkrus012345 opened the pull request:
## 프로젝트 자바 버전 설정
자바 21과 자바 17 중 고민 끝에 17을 선택했습니다.
자바 21의 가상 스레드는 자바 17에 비해 대규모 동시성 처리에 유리한 장점이 있지만 현재 프로젝트는 그런 수준의 동시성 처리가 필요한 구조는 아니라고 판단했습니다.
따라서 가장 널리 사용되고 있으며 안정성이 검증되어 있는 저버 17를 적용했습니다.
향후 동시성 처리 요구가 명확해지고 관련 기술 도입이 필요해질 경우 자바 21로의 마이그레이션을 고려하겠습니다.

tlqkrus012345 committed:
Feat #1 : 프로젝트 초기화

tlqkrus012345 committed:
Test #1 : 테스트 컨트롤러 생성

f-lab-martin reviewed (approved):
# 1. 코드 포맷팅
intellij를 사용하신다면 Actions on Save 설정을 통해 optimize import, reformat code를 체크하시고 사용하시면 편합니다. 
reformat code는 google java style 가이드를 설정하시고 적용해보시길 추천드립니다.

참고 
- https://hbase.tistory.com/451
- https://blog.naver.com/seek316/223306071687

# 2. 커밋 컨벤션
- 커밋 컨벤션을 맞추시려고 시도하신 것으로 보입니다. 찾아보시면 많이 쓰이는 컨벤션들이 있는데 찾아보시고 적용해보시면 좋을 것 같습니다. 아래 내용은 참고차 같이 공유드립니다.

## 커밋 컨벤션 정리

커밋 컨벤션(Commit Convention)은 협업과 이력 관리의 효율성을 위해 커밋 메시지 작성 규칙을 정해두는 것입니다. 대표적으로 **Conventional Commits** 규칙이 많이 사용됩니다.

---

### 1. 커밋 메시지 구조

일반적으로 커밋 메시지는 세 부분으로 나눕니다:

```
<type>[optional scope]: <subject>

[body]

[footer]
```

- **type**: 커밋의 목적(기능 추가, 버그 수정 등)
- **subject(제목)**: 50자 이내, 첫 글자 대문자, 명령문, 마침표 금지
- **body(본문)**: 선택, 72자 이내, 무엇을/왜 변경했는지 설명
- **footer(꼬리말)**: 선택, 이슈 번호 등 참조

---

### 2. 커밋 타입 예시

| 타입      | 설명                                   |
|-----------|----------------------------------------|
| feat      | 새로운 기능 추가                       |
| fix       | 버그 수정                              |
| refactor  | 리팩토링(기능 변화 없는 코드 개선)      |
| style     | 코드 스타일, 포맷팅, 세미콜론 등       |
| docs      | 문서, 주석 등 수정                     |
| test      | 테스트 코드 추가/수정                  |
| chore     | 기타 변경(빌드, 패키지, 환경설정 등)   |
| build     | 빌드 관련 파일 수정                    |
| ci        | CI 설정 변경                           |
| perf      | 성능 개선                              |
| remove    | 파일 삭제                              |
| rename    | 파일/폴더명 변경                       |
| init      | 프로젝트 초기화                        |
| comment   | 주석 추가/수정                         |
| design    | UI, CSS 등 디자인 변경                 |

---

### 3. 커밋 메시지 작성 규칙

- 제목과 본문 사이 한 줄 띄우기
- 제목은 50자 이내, 본문은 72자 이내로 작성
- 제목은 명령문(현재형), 첫 글자 대문자, 마침표 금지
- 본문에는 "무엇을"과 "왜"에 초점, "어떻게"는 생략 가능
- 꼬리말에는 이슈 번호, 관련 트래커 등 명시 (예: `Resolves: #10`)
- 한글/영문 모두 가능하나, 팀 규칙에 따름

---

### 4. 커밋 메시지 예시

```
Feat: 회원가입 기능 추가

- 회원가입 API 엔드포인트 구현
- 입력값 유효성 검증 추가

Resolves: #12
```

```
Fix: 게시판 글 조회 오류 수정

- BoardController 글 조회 메소드 userService > boardService로 변경
```

---

### 5. 커밋 템플릿 적용

- 템플릿 파일(`gitmessage.txt`)을 만들어 규칙을 미리 넣고,  
  `git config --global commit.template <경로>`로 적용하면 일관성 유지에 도움이 됩니다.

---

### 6. 참고

- 커밋 컨벤션을 지키면 협업 시 변경 이력 파악, 자동화 도구 연동, 릴리즈 노트 생성 등이 쉬워집니다.
- 팀 상황에 맞게 일부 규칙은 조정할 수 있습니다.

---

**요약:**  
커밋 컨벤션은 커밋 메시지의 일관성과 명확성을 위해 작성 규칙을 정하는 것입니다.  
대표적으로 type, subject, body, footer 구조와 명령문, 50자 제한, 이슈 번호 명시 등이 포함됩니다.  
주요 타입(feat, fix, refactor 등)을 활용해 커밋 목적을 명확히 드러내는 것이 핵심입니다.

tlqkrus012345 committed:
style: 깃 커밋 템플릿 작성

- 커밋 메시지의 일관성과 명확성을 위해 커밋 템플릿을 작성
- 팀원과 공유하기 위해 해당 파일은 깃 허브에 파일 등록

Resolves: #1
